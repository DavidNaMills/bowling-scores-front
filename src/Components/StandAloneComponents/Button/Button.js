import React from 'react';
import classes from './Button.module.scss';



const Button = ({ click = () => { }, label, isDisabled = false, type = 'default' }) => {
    let style = [classes.button];

    if (isDisabled) {
        style = style.concat(classes.button__disabled);
    } else {

        switch (type.toLowerCase()) {
            case 'warning':
                style = style.concat(classes.button__warning);
                break;
            case 'danger':
                style = style.concat(classes.button__danger);
                break;
            default:
                style = style.concat(classes.button__default);
                break;
        }
    }

    return (
        <button onClick={click} disabled={isDisabled} className={style.join(' ')}>{label}</button>
    );
};

export default Button;