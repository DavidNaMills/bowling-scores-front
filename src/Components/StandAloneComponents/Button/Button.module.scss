@import '../../../styles/colors.scss';


.button{
    border: none;
    font-size: 16px;
    padding: 10px;
    min-width: 100px;
    font-weight: 550;

    &__disabled{
        background-color: $btn_disabled;
    }
    
    &__isFull{
        width: 100%;
    }

    &__default{
        background-color: $btn_default_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_default_bkg_active_hover;
        }
        &:active{
            background-color: $btn_default_bkg_active;
        }
    }

    &__warning{
        background-color: $btn_warning_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_warning_bkg_active_hover;
        }
        &:active{
            background-color: $btn_warning_bkg_active;
        }    
    }
    
    &__danger{
        background-color: $btn_danger_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_danger_bkg_active_hover;
        }
        &:active{
            background-color: $btn_danger_bkg_active;
        }    
    }

    &__lightGreen{
        background-color: $btn_lightGreen_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_lightGreen_bkg_active;
        }
        &:active{
            background-color: $btn_lightGreen_bkg_active_hover;
        }    
    }

    &__darkGreen{
        background-color: $btn_darkGreen_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_darkGreen_bkg_active;
        }
        &:active{
            background-color: $btn_darkGreen_bkg_active_hover;
        }    
    }

    &__blue{
        background-color: $btn_blue_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_blue_bkg_active;
        }
        &:active{
            background-color: $btn_blue_bkg_active_hover;
        }    
    }

    &__lightBlue{
        background-color: $btn_lightBlue_bkg;
        color: $btn_font_clr;
        
        &:hover{
            background-color: $btn_lightBlue_bkg_active;
        }
        &:active{
            background-color: $btn_lightBlue_bkg_active_hover;
        }    
    }

    &__darkBlue{
        background-color: $btn_darkBlue_bkg;
        color: $btn_font_clr_light;
        
        &:hover{
            background-color: $btn_darkBlue_bkg_active;
        }
        &:active{
            background-color: $btn_darkBlue_bkg_active_hover;
        }    
    }

    &__lightRed{
        background-color: $btn_lightRed_bkg;
        color: $btn_font_clr;
        
        &:hover{
            color: $btn_font_clr_light;
            background-color: $btn_lightRed_bkg_active;
        }
        &:active{
            background-color: $btn_lightRed_bkg_active_hover;
        }    
    }

    &__lightYellow{
        background-color: $btn_lightYellow_bkg;
        color: $btn_font_clr;
        
        &:hover{
            color: $btn_font_clr_light;
            background-color: $btn_lightYellow_bkg_active;
        }
        &:active{
            background-color: $btn_lightYellow_bkg_active_hover;
        }    
    }



}