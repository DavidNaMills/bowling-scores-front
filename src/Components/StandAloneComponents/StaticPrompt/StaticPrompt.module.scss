.staticPrompt{
    background: rgb(162,255,150);
    background: -moz-linear-gradient(180deg, rgba(162,255,150,1) 37%, rgba(18,255,235,1) 100%);
    background: -webkit-linear-gradient(180deg, rgba(162,255,150,1) 37%, rgba(18,255,235,1) 100%);
    background: linear-gradient(180deg, rgba(162,255,150,1) 37%, rgba(18,255,235,1) 100%);
    filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#a2ff96",endColorstr="#12ffeb",GradientType=1);

    display: flex;
    justify-content: center;
    padding-bottom: 20px;
    margin-bottom: 10px;

    &__btn{
        display: flex;
        justify-content: space-between;
        &_hf{
            width: 40%;
        }

        &_fl{
            width: 100%;
        }
    }
}