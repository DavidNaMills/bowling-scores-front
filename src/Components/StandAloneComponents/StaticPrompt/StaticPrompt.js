import React from 'react';
import Button from '../Button/Button';
import popup from '../Popup/Popup.module.scss';
import classes from './StaticPrompt.module.scss';

const StaticPrompt = ({ message, button1 = null, button2 = null, close=null }) => (
    <div className={classes.staticPrompt}>
        {close &&
            <div className={popup.popup__closeBar}>
                <p onClick={close} className={popup.popup__closeBar_cls}>CLOSE</p>
            </div>
        }
        <div>
            <p>{message}</p>
            <div className={classes.staticPrompt__btn}>
                {button1 && <div className={button1&&button2
                    ? classes.staticPrompt__btn_hf
                    : classes.staticPrompt__btn_fl
                }><Button isFull {...button1} /></div>}
                {button2 && <div className={button1&&button2
                    ? classes.staticPrompt__btn_hf
                    : classes.staticPrompt__btn_fl
                }><Button isFull {...button2} /></div>}
            </div>
        </div>
    </div>
);

export default StaticPrompt;