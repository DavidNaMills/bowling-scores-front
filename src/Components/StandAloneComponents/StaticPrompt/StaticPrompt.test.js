import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import StaticPrompt from './StaticPrompt';
import Button from '../Button/Button';

const message = 'Login or signup to save your game!';



describe('<LoginPrompt/> test suite', () => {
     it('snapshot tests', () => {
        const testMesg = 'This is a test message';
        const component = renderer.create(<StaticPrompt message={testMesg} />);
        const app = component.toJSON();
        expect(app).toMatchSnapshot();
    });



    it('should contain a <p> element with text', () => {
        const wrapper = mount(
            <StaticPrompt
                message={message}
            />
        );
        const msg = wrapper.find('p');
        expect(msg.length).toBe(1);
        expect(msg.text()).toEqual(message);
    });


    it('should display 1 button', () => {
        const button1 = {
            click: jest.fn(),
            label: 'Signup',
            type: 'darkGreen'
        };
        
        const wrapper = mount(
            <StaticPrompt
                message={message}
                button1={button1}
            />
        );
        const btn1 = wrapper.find(Button);
        expect(btn1.length).toBe(1);
        btn1.at(0).simulate('click');
        expect(button1.click).toHaveBeenCalledTimes(1);
    });
    
    it('should display a close bar', ()=>{
        const close = jest.fn();
        const wrapper = mount(<StaticPrompt close = {close}/>);
        const cls = wrapper.find('p');

        expect(cls.length).toBe(2);
        expect(cls.at(0).text()).toEqual('CLOSE');
        cls.at(0).simulate('click');
        expect(close).toHaveBeenCalledTimes(1);
    });

    it('should display 2 buttons', () => {
        const button1 = {
            click: jest.fn(),
            label: 'Signup',
            type: 'darkGreen'
        };
        
        const button2 = {
            click: jest.fn(),
            label: 'Signup',
            type: 'darkBlue'
        };

        const wrapper = mount(
            <StaticPrompt
                message={message}
                button1={button1}
                button2={button2}
            />
        );
        const btn = wrapper.find(Button);
        expect(btn.length).toBe(2);
        btn.at(0).simulate('click');
        btn.at(1).simulate('click');
        expect(button1.click).toHaveBeenCalledTimes(1);
        expect(button2.click).toHaveBeenCalledTimes(1);

    });
});