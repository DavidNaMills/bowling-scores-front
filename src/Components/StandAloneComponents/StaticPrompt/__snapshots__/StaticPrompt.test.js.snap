// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPrompt/> test suite snapshot tests 1`] = `
<div
  className="staticPrompt"
>
  <div>
    <p>
      This is a test message
    </p>
    <div
      className="staticPrompt__btn"
    />
  </div>
</div>
`;
